### PR TITLE
test: fix gomock controller finish

### PR DESCRIPTION
**Pull request by dadamu**
## Description

This PR fixes the gomock controller finish by moving it to the `TearDownTest` phase, or it cannot find the missing calls of the function we setup on expected keepers.

It also fixes the tests having wrong setting on the expected keepers.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [ ] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/posts/keeper/genesis_test.go [hunk 1]
```diff
@@ -291,34 +291,12 @@ func (suite *KeeperTestSuite) TestKeeper_ExportGenesis() {
 func (suite *KeeperTestSuite) TestKeeper_ImportGenesis() {
 	testCases := []struct {
 		name  string
-		setup func()
```
Comment (dadamu) [inline x/posts/keeper/genesis_test.go:294]: ImportGenesis does not require any expected keepers setting.

File: x/subspaces/ante/tx_checker_test.go [hunk 2]
```diff
@@ -100,16 +100,6 @@ func (suite *AnteTestSuite) TestCheckTxFeeWithSubspaceMinPrices() {
...
 			name: "manage subspace tx with subspace fees returns error",
-			setup: func() {
-				suite.sk.EXPECT().
-					GetSubspace(gomock.Any(), uint64(1)).
-					Return(
-						types.Subspace{
-							AdditionalFeeTokens: sdk.NewCoins(sdk.NewCoin("minttoken", sdk.NewInt(1))),
-						},
-						true,
-					)
```
Comment (dadamu) [inline x/subspaces/ante/tx_checker_test.go:111]: manage subspace transaction does not check the subspaces additional fee tokens.

File: x/posts/keeper/msg_server_test.go [hunk 3]
```diff
@@ -1491,14 +1491,6 @@ func (suite *KeeperTestSuite) TestMsgServer_RemovePostAttachment() {
 					"cosmos13t6y2nnugtshwuy0zkrq287a95lyy8vzleaxmd",
 					subspacestypes.NewPermission(types.PermissionModerateContent),
 				).Return(false)
-
-				suite.sk.EXPECT().HasPermission(
```
Comment (dadamu) [inline x/posts/keeper/msg_server_test.go:1495]: It does not check the `EditOwnContent` permission when post owner does not match the message sender.

File: x/posts/keeper/msg_server_test.go [hunk 4]
```diff
@@ -669,7 +669,7 @@ func (suite *KeeperTestSuite) TestMsgServer_EditPost() {
 					types.REPLY_SETTING_EVERYONE,
 					time.Date(2020, 1, 1, 12, 00, 00, 000, time.UTC),
 					nil,
-					"cosmos1r9jamre0x0qqy562rhhckt6sryztwhnvhafyz4",
+					"cosmos13t6y2nnugtshwuy0zkrq287a95lyy8vzleaxmd",
```
Comment (dadamu) [inline x/posts/keeper/msg_server_test.go:672]: We should ensure the the owner of post stored is the same as the user without `EditOwnContent` permission in this test

File: x/posts/keeper/msg_server_test.go [hunk 5]
```diff
@@ -1170,7 +1170,7 @@ func (suite *KeeperTestSuite) TestMsgServer_AddPostAttachment() {
 					gomock.Any(),
 					uint64(1),
 					uint32(0),
-					"cosmos1r9jamre0x0qqy562rhhckt6sryztwhnvhafyz4",
+					"cosmos13t6y2nnugtshwuy0zkrq287a95lyy8vzleaxmd",
```
Comment (dadamu) [inline x/posts/keeper/msg_server_test.go:1173]: We should ensure the the owner of post stored is the same as the sender without EditOwnContent permission in this test.

File: x/posts/keeper/msg_server_test.go [hunk 6]
```diff
@@ -1189,7 +1189,7 @@ func (suite *KeeperTestSuite) TestMsgServer_AddPostAttachment() {
 					types.REPLY_SETTING_EVERYONE,
 					time.Date(2020, 1, 1, 12, 00, 00, 000, time.UTC),
 					nil,
-					"cosmos1r9jamre0x0qqy562rhhckt6sryztwhnvhafyz4",
+					"cosmos13t6y2nnugtshwuy0zkrq287a95lyy8vzleaxmd",
```
Comment (dadamu) [inline x/posts/keeper/msg_server_test.go:1192]: As above

File: x/posts/keeper/msg_server_test.go [hunk 7]
```diff
@@ -3282,7 +3292,6 @@ func (suite *KeeperTestSuite) TestMsgServer_RequestPostOwnerTransfer() {
 func (suite *KeeperTestSuite) TestMsgServer_CancelPostOwnerTransfer() {
 	testCases := []struct {
 		name        string
-		setup       func()
```
Comment (dadamu) [inline x/posts/keeper/msg_server_test.go:3285]: CancelPostOwnerTransfer does not require expected keepers setting

File: x/posts/keeper/msg_server_test.go [hunk 8]
```diff
@@ -3649,7 +3639,6 @@ func (suite *KeeperTestSuite) TestMsgServer_AcceptPostOwnerTransfer() {
 func (suite *KeeperTestSuite) TestMsgServer_RefusePostOwnerTransfer() {
 	testCases := []struct {
 		name        string
-		setup       func()
```
Comment (dadamu) [inline x/posts/keeper/msg_server_test.go:3652]: As above, `RefusePostOwnerTransfer` does not require expected keepers setting

**Review by RiccardoM — APPROVED** [review]
Great job! :100:

**Pull request merged**